### PR TITLE
Fix tracing

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Enabled tracing for natss dispatcher. Using sample rate 0.1 for now but that can be configured as per need.

**Related issue(s)**

See #6226

### CODE REVIEW

File: resources/event-bus/charts/event-publish-service/templates/_helpers.tpl [hunk 1]
```diff
@@ -20,6 +20,7 @@ app.kubernetes.io/name: {{ template "event-publish-service.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
 app.kubernetes.io/part-of: kyma
+app: {{ template "event-publish-service.name" . }}
```
Comment (antoineco) [inline resources/event-bus/charts/event-publish-service/templates/_helpers.tpl]: We had a PR some time ago to explicitly avoid those labels. Why are they back?

File: resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/pre-upgrade-job.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Release.Name }}-pre-upgrade
+  annotations:
+    "helm.sh/hook": "pre-upgrade"
```
Comment (antoineco) [inline resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/pre-upgrade-job.yaml]: That's JSON notation. YAML has no quotes.

File: resources/event-bus/charts/event-publish-service/templates/_helpers.tpl (hunk 1 above)
Comment (sayanh) [reply resources/event-bus/charts/event-publish-service/templates/_helpers.tpl]: These are not needed hence I removed it.

File: resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/pre-upgrade-job.yaml (hunk 2 above)
Comment (sayanh) [reply resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/pre-upgrade-job.yaml]: True that hence removed except the weights where we wanna force them to be strings.

**Comment by sayanh** [comment]
@antoineco 
> Could you please add one sentence to the issue description explaining why using a sample rate value of 0.1 fixes the missing traces? The README says 0.1 is not enough, so I'm a bit confused.

We are enabling traces as you see in the config map and using 0.1 as the sampling rate. Just using "sample rate value of 0.1" doesn't fix the issue. This is good for a start. Meaning, if a trace is sampled, then the complete trace is there otherwise the complete trace is missing. Documentation talks about what needs to be done if one needs more trace sampling.

File: docs/tracing/10-01-traces-troubleshooting.md [hunk 3]
```diff
@@ -5,11 +5,20 @@ type: Troubleshooting
 
 ## Jaeger shows only a few traces
 
-The current Istio Pilot settings define the trace sampling rate at `1.0`, where `100` is the maximum value. This means that only 1 out of 100 requests is sent to Jaeger for trace recording. To change this system behavior, run:
+1. The current Istio Pilot settings define the trace sampling rate at `1.0`, where `100` is the maximum value. This means that only 1 out of 100 requests is sent to Jaeger for trace recording. To change this system behavior, run:
```
Comment (bszwarc) [inline docs/tracing/10-01-traces-troubleshooting.md]:
```suggestion
Jaeger showing just a few traces may be caused by the following settings:
1. Istio Pilot sets the trace sampling rate at `1.0`, where `100` is the maximum value. This means that only 1 out of 100 requests is sent to Jaeger for trace recording. To change this system behavior, run:
```
Let's add a short introductory sentence to the points.

File: docs/tracing/10-01-traces-troubleshooting.md [hunk 4]
```diff
@@ -5,11 +5,20 @@ type: Troubleshooting
...
 
 ```bash
 kubectl -n istio-system edit deploy istio-pilot
 ```
 Set the **traceSampling** parameter to a desired value, such as `60`.
 
 >**NOTE**: Using a very high value may affect Istio's performance and stability.
+
+2. The current Knative trace sampling rate is also at `0.1` where `1` is the maximum value. To have complete trace recordings, run:
```
Comment (bszwarc) [inline docs/tracing/10-01-traces-troubleshooting.md]:
```suggestion
2. The current Knative trace sampling rate is at `0.1` where `1` is the maximum value. To have complete trace recordings, run:
```

File: resources/knative-eventing/values.yaml [hunk 5]
```diff
@@ -1,2 +1,8 @@
 global:
   isLocalEnv: false
+
+knative-eventing:
+  tracing:
+    enable: true
```
Comment (k15r) [inline resources/knative-eventing/values.yaml]: is there a way to get this value from kymas istio configuration? It should be the same value across all kyma

File: resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/natss.yaml [hunk 6]
```diff
@@ -162,7 +163,9 @@ spec:
         - port: 80
           protocol: TCP
           targetPort: 8080
+          name: http
     selector:
+        app: natss-ch-dispatcher
```
Comment (k15r) [inline resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/natss.yaml]: Do we really need the app in the selector? The selector should also apply to this pod without relying on the app-label.
If i understand it correctly we put app here just for proper trace naming.

File: resources/knative-eventing/values.yaml [hunk 7]
```diff
@@ -1,2 +1,8 @@
...
+    collectorUrl: http://zipkin.kyma-system.svc.cluster.local:9411/api/v2/spans
```
Comment (sayanh) [inline resources/knative-eventing/values.yaml]: Add a comment on why?

File: resources/knative-eventing/values.yaml (hunk 5 above)
Comment (sayanh) [reply resources/knative-eventing/values.yaml]: Definitely, there is a way by using "overrides". But that will bloat the PR scope IMO and we have to holistically solve it for all the components e.g. publish-service, console-backend-service. How about solving it in a different issue?

File: resources/knative-eventing/values.yaml (hunk 7 above)
Comment (sayanh) [reply resources/knative-eventing/values.yaml]: Comment is added.

File: resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/natss.yaml (hunk 6 above)
Comment (sayanh) [reply resources/knative-provisioner-natss/charts/knative-provisioner-natss/templates/natss.yaml]: No, we don't need selector. Hence removed it as well as the pre-upgrade hook.

**Pull request merged**